server/mem_agent/tools.py

```
@@ -0,0 +1,359 @@
+        with open(temp_file_path, "w") as f:
+            f.write(content)
+        
+        if check_size_limits(temp_file_path):
+            # Move the content to the final destination
+            with open(file_path, "w") as f:
+                f.write(content)
+            os.remove(temp_file_path)
+            return True
+        else:
+            os.remove(temp_file_path)
+            raise Exception(f"File {file_path} is too large to create")
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)